SignalMessaging/environment/AppSetup.m

```diff
@@ -46,7 +51,18 @@ + (void)setupEnvironment:(CallMessageHandlerBlock)callMessageHandlerBlock
         [NSKeyedUnarchiver setClass:[OWSUserProfile class] forClassName:[OWSUserProfile collection]];
         [NSKeyedUnarchiver setClass:[OWSDatabaseMigration class] forClassName:[OWSDatabaseMigration collection]];
 
-        [OWSStorage setupStorage];
+        [OWSStorage setupStorageWithMigrationBlock:^() {
+            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+                // Don't start database migrations until storage is ready.
+                [VersionMigrations performUpdateCheckWithCompletion:^() {
+                    OWSAssertIsOnMainThread();
+
+                    migrationCompletion();
+
+                    backgroundTask = nil;
+                }];
+            });
+        }];
         [[Environment current].contactsManager startObserving];
```

Comment:
Old code, but do we really want contactsManager to start observing here? Seems like we wouldn't want it to do anything until **after** the db is up and migrations have completed.

In this case, I think the consequences are benign because it just clears an entry from the avatar cache, but it would be *really* easy to inadvertently add some code which opens a write transaction into `ContactsManager.otherUsersProfileWillChange` without realizing it can be called before the app is ready.

Reply:
Great point, moved to when app becomes ready.  And I'm _not_ going to start this in the SAE.